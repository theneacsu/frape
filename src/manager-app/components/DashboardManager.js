import React from 'react'
import { connect } from 'react-redux'
import { Link } from 'react-router-dom'
import getVisibleTodos from '../../selectors/todo-app/todos'

export const DashboardManager = props => (
  <div className="dashboard-box">
    <h1>{props.userInfo.displayName}</h1>
    <img src={props.userInfo.photoURL} alt="User profile phote" />
    <p>
      You have <Link to="/todos">{props.todosLeft} todo(s)</Link> left.
    </p>
  </div>
)

const mapStateToProps = state => ({
  userInfo: state.auth,
  todosLeft: getVisibleTodos(state.todos, state.todosFilters).filter(
    todo => !todo.completed
  ).length
})

export default connect(mapStateToProps)(DashboardManager)
