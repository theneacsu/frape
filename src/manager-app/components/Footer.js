import React from 'react'

const Footer = () => (
  <div className="footer">
    Made with{' '}
    <img className="footer-icon" src="images/like.png" alt="heart icon" /> and{' '}
    <img className="footer-icon" src="images/coffee.png" alt="coffee icon" /> by
    Neacsu Alexandru
  </div>
)

export default Footer
