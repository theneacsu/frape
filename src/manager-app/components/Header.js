import React from 'react'
import { connect } from 'react-redux'
import { NavLink } from 'react-router-dom'
import { startLogout } from '../../actions/auth/auth'

export const Header = props => (
  <div className="box-header">
    <div className="header-container">
      <nav className="box-header__nav">
        <NavLink
          className="menu-item"
          to="/dashboard"
          activeClassName="is-active"
        >
          Dashboard
        </NavLink>
        <NavLink className="menu-item" to="/todos" activeClassName="is-active">
          Todos
        </NavLink>
      </nav>
      <button className="logout-button" onClick={props.startLogout}>
        Logout
      </button>
    </div>
  </div>
)

const mapDispatchToProps = dispatch => ({
  startLogout: () => dispatch(startLogout())
})

export default connect(
  undefined,
  mapDispatchToProps
)(Header)
