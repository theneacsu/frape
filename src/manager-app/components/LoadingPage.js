import React from 'react'

const LoadingPage = () => (
  <div class="lds-heart">
    <div />
  </div>
)

export default LoadingPage
