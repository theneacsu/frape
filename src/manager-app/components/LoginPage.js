import React from 'react'
import { connect } from 'react-redux'
import { startLogin } from '../../actions/auth/auth'

export const LoginPage = props => (
  <div className="box-layout">
    <div className="box-layout__box">
      <h1 className="box-layout__title">Holistic Integration</h1>
      <h2 className="box-layout__subtitle">
        Increase your performance, focus and productivity
      </h2>
      <button className="button" onClick={props.startLogin}>
        Get started with Google
      </button>
    </div>
  </div>
)

const mapDispatchToProps = dispatch => ({
  startLogin: () => dispatch(startLogin())
})

export default connect(
  undefined,
  mapDispatchToProps
)(LoginPage)
