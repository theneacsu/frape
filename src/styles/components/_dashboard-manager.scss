.dashboard-box {
  background-color: rgb(229, 243, 241);
  text-align: center;
  min-height: 100vh;
  h1 {
    margin: 0;
    font-size: 2.5rem;
    padding: 3rem $m-size;
  }

  img {
    width: 20rem;
    height: 20rem;
    border-radius: 50%;
    display: block;
    margin: 0 auto;
  }

  a {
    text-decoration: none;
    color: #4cb69f;
    font-weight: bold;
    font-size: 2rem;
  }

  p {
    font-size: 2rem;
  }
}
