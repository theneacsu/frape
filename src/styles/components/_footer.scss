.footer {
  text-align: center;
  font-size: 1.7rem;
  font-style: italic;
  font-weight: 100;
  padding: 1rem;
  background-color: #4cb69f;
  color: white;
}

.footer-icon {
  height: 1.7rem;
}
