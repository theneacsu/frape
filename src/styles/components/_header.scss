.box-header {
  text-align: center;
  margin: 0;
  background-color: #4cb69f;
}

.header-container {
  display: flex;
  justify-content: space-between;
}

.box-header__nav {
  display: flex;
}

.box-header a {
  padding: 1.3rem 2rem;
  margin: 0;
}

.menu-item {
  text-decoration: none;
  color: white;
  letter-spacing: 0.12rem;
  font-weight: bold;
}

.logout-button {
  border: none;
  outline: none;
  background-color: #4cb69f;
  font-weight: bold;
  color: white;
  padding: 1rem 2rem;
}

.logout-button:hover,
.menu-item:hover {
  cursor: pointer;
}

@media (min-width: 600px) {
  .header-container {
    width: 80%;
    max-width: 800px;
    margin: auto;
  }
}
