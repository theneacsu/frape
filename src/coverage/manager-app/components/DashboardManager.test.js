import React from 'react'
import { shallow } from 'enzyme'
import { DashboardManager } from '../../../manager-app/components/DashboardManager'

test('Should render correctly the DashboardManager component', () => {
  const user = {
    uid: '123abc',
    photoURL:
      'https://lh4.googleusercontent.com/-3HA5dfxwrYU/AAAAAAAAAAI/AAAAAAAAEJ4/q6fYoZQpeEY/photo.jpg',
    displayName: 'Bradley Cooper',
    email: 'test@test.com'
  }
  const wrapper = shallow(<DashboardManager userInfo={user} todosLeft={3} />)
  expect(wrapper).toMatchSnapshot()
})
