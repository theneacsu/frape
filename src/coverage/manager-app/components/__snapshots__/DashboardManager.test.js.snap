// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render correctly the DashboardManager component 1`] = `
<div
  className="dashboard-box"
>
  <h1>
    Bradley Cooper
  </h1>
  <img
    alt="User profile phote"
    src="https://lh4.googleusercontent.com/-3HA5dfxwrYU/AAAAAAAAAAI/AAAAAAAAEJ4/q6fYoZQpeEY/photo.jpg"
  />
  <p>
    You have 
    <Link
      replace={false}
      to="/todos"
    >
      3
       todo(s)
    </Link>
     left.
  </p>
</div>
`;
